### ISSUE TITLE
Simplify training of the memory module

### ISSUE THREAD

**Issue by RPegoud** [RPegoud/Hercules#1]
In the paper, the memory module is trained using an associative scan which is kind of a pain to write and optimise.
For now, we should probably start with basic gradient descent with momentum and forget gate. I figure the difference in performance will be minimal compared to the cost of inference for larger models.
